### ISSUE TITLE
Performance comparison: YOLO11s/12s outperforming YOLO26s across multiple datasets

### ISSUE THREAD

**Issue by GKXKX123456** [ultralytics/ultralytics#23599]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I have been conducting benchmark tests using models ranging from YOLOv8s to YOLO26s across 7 different datasets. During my testing, I observed some interesting results regarding model accuracy (mAP).

Observations:

YOLO11s and YOLO12s consistently achieved the highest mAP levels across the tested datasets.

Surprisingly, YOLO26s performed slightly lower than both YOLO11s and YOLO12s, despite the architectural differences.

Test Environment:

Models tested: YOLOv8s through YOLO26s.

Dataset count: 7 diverse datasets.

Primary Metric: mAP (Mean Average Precision).

Version: 8.4.11

I was wondering if the community or the maintainers have conducted similar precision validations across these specific versions? I am curious to know if this trend aligns with your official benchmarks or if there are specific configurations where YOLO26s is expected to show its advantages.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @GKXKX123456, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Thanks for sharing your benchmarks across YOLO11s/YOLO12s/YOLO26s 📊—this kind of comparison is really helpful for the community. This is an automated response 🤖, and an Ultralytics engineer will also assist soon.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it (especially if you suspect a version/config regression).

If this is a custom training ❓ Question, please provide as much information as possible (so we can compare apples-to-apples), including:
-Exact model commands/configs used for each run (CLI or Python)
-Dataset YAML(s) (or the relevant parts: nc/names/train/val paths)
-Training logs (per-epoch metrics) and final validation outputs
-Any hyperparameters you changed from defaults (imgsz, batch, epochs, augmentations, optimizer, lr schedule, etc.)
-Hardware + OS + CUDA/PyTorch versions
-And please confirm you’re following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) ✅

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
You can get higher mAP with YOLO26s by using NMS based mode using `end2end=False` during validation.

The NMS free mode has lower mAP but faster speed. While NMS based mode has higher mAP but slower speed. It's a tradeoff. You can use either depending on your requirements.

**Comment by GKXKX123456** [comment]
> You can get higher mAP with YOLO26s by using NMS based mode using `end2end=False` during validation.
> 
> The NMS free mode has lower mAP but faster speed. While NMS based mode has higher mAP but slower speed. It's a tradeoff. You can use either depending on your requirements.

Thanks for the advice. Turning off end-to-end in the val stage did improve accuracy on some datasets. But I'm seeing a weird regression with YOLOv26: on one dataset, it shows a 15% mAP drop after 200 epochs (whereas v10-v12 are fine). Curiously, at only 20 epochs, the drop is much smaller—around 5%. Any idea why longer training is hurting performance here?

**Comment by Y-T-G** [comment]
How large is your dataset? It seems like overfitting

**Comment by GKXKX123456** [comment]
> How large is your dataset? It seems like overfitting

For the experiment where this occurred, I was training the model on a dataset with 11 classes and 2,500 images.

**Comment by glenn-jocher** [comment]
With 2,500 images / 11 classes, a 200-epoch run can definitely overfit; first, can you confirm you’re comparing `runs/.../weights/best.pt` (selected by val fitness) and not `last.pt`? If `best` peaks early, the simplest fix is fewer epochs or `patience`-based early stopping, plus a bit more regularization (e.g. higher `weight_decay`)—the general checklist in the [performance metrics deep dive](https://docs.ultralytics.com/guides/yolo-performance-metrics/) is a good reference.

If you paste your exact `yolo train` command, the epoch where mAP peaks/drops (from `results.csv`), and your `yolo checks` output, we can tell whether it’s pure overfit vs a config mismatch:

```bash
yolo checks
```

**Comment by MasIgor** [comment]
> You can get higher mAP with YOLO26s by using NMS based mode using `end2end=False` during validation.
> 
> The NMS free mode has lower mAP but faster speed. While NMS based mode has higher mAP but slower speed. It's a tradeoff. You can use either depending on your requirements.

how does this affect onnx files? If i export my model to onnx to use in C++ code, do i have to set this to false as well somehow?
or does it have to be done during export?

**Comment by glenn-jocher** [comment]
`end2end=False` during `val` only affects evaluation/postprocessing (it doesn’t change the trained weights); for ONNX it’s an **export-time** choice: export NMS-free (`end2end=True`/default) if your C++ pipeline will consume the end-to-end outputs, or export the traditional NMS-style head with `end2end=False` (and optionally bake NMS into the ONNX with `nms=True`) as documented in the [Export mode arguments](https://docs.ultralytics.com/modes/export/#arguments).

```bash
yolo export model=path/to/best.pt format=onnx end2end=False nms=True
```

**Comment by banu4prasad** [comment]
> Dataset count: 7 diverse datasets.

could u plz tell us the datasets which ur using, with number of classes and dataset split train/val ?

**Comment by glenn-jocher** [comment]
Yes, please share the dataset names, class counts, train/val split, and the exact `train`/`val` settings for one representative run; our official [YOLO11 vs YOLO26 benchmark](https://docs.ultralytics.com/compare/yolo11-vs-yolo26/) is on COCO at `imgsz=640`, where `YOLO26s` is slightly ahead of `YOLO11s`, so a consistent gap on your 7 datasets would likely be dataset- or config-specific.
